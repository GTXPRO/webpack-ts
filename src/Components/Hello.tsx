/**
 * @description   ¯\(ツ)/¯
 * @since         Saturday, 6 26th 2021, 22:36:04 pm
 * @author        Nguyễn Đạt <nguyendinhdat@getflycrm.com>
 * @copyright     Copyright (c) 2021, GETFLY VN TECH.,JSC
 * -----
 * Change Log: <press Ctrl + alt + c write changelog>
 */


import React, { FC } from 'react';

interface HelloProps {
	name?: string,
	age: number,
};

const Hello: FC<HelloProps> = ({ name, age }) => (
	<div>
		Name: {name} <br />
		Age: {age}
	</div>
);

export default Hello;
