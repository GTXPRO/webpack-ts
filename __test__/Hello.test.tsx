/**
 * @description   ¯\(ツ)/¯
 * @since         Wednesday, 8 25th 2021, 8:36:45 am
 * @author        Nguyễn Đạt <nguyendinhdat@getflycrm.com>
 * @copyright     Copyright (c) 2021, GETFLY VN TECH.,JSC
 * -----
 * Change Log: <press Ctrl + alt + c write changelog>
 */


import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import Hello from './../src/Components/Hello';
// import TestRenderer from 'react-test-renderer';

describe('Test Component Hello By Text', () => {
  it('String: "Trần Văn Tú" contains correct text', () => {
    render(<Hello age={11} name="Trần Văn Tú" />);
    expect(screen.getByText(/Trần Văn Tú/i)).toBeInTheDocument();
  });

  it('String: "TranChemStore.com" contains correct text', () => {
    render(<Hello age={11} name="TranChemStore.com" />);
    expect(screen.getByText(/TranChemStore\.com/i)).toBeInTheDocument();
  });

  it('Snapshot: "Hello" contains correct text', () => {
    const { asFragment } = render(<Hello age={11} name="Hello" />);
    expect(asFragment()).toMatchSnapshot();
  });

  afterEach(cleanup);
});
