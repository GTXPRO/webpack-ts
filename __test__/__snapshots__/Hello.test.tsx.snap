// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component Hello By Text Snapshot: "Hello" contains correct text 1`] = `
<DocumentFragment>
  <div>
    Name: Hello 
    <br />
    Age: 11
  </div>
</DocumentFragment>
`;
